testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/vault/CustomVaultInModuleTestCase.java

```diff
@@ -113,10 +113,9 @@ public void testStandardRemoteExpressionResolution() throws Exception {
         attrValue = response.get("result", "attribute");
         Assert.assertEquals(response.toString(), ModelType.EXPRESSION, attrValue.getType());
         Assert.assertEquals(response.toString(), "${VAULT::Testing::Stuff::thing}", attrValue.asString());
-        // TODO uncomment when WFCORE-5305 is resolved
-        //responseHeaders = response.get("response-headers").asString();
-        //Assert.assertTrue(response.toString(), responseHeaders.contains("${VAULT::Testing::Stuff::thing}"));
-        //Assert.assertTrue(response.toString(), responseHeaders.contains("WFLYCTL0479"));
+        responseHeaders = response.get("response-headers").asString();
+        Assert.assertTrue(response.toString(), responseHeaders.contains("${VAULT::Testing::Stuff::thing}"));
+        Assert.assertTrue(response.toString(), responseHeaders.contains("WFLYCTL0479"));
```

Comment:
To fully test this you'll need to add another resource level to this test extension and put an expression there as well. And then read recursively and confirm the warn for the child resource propagates.

Reply:
Please do let me know if more changes needed in the test case except for that scenario which is started in zulip chat.

Reply:
One scenario is still left which I have asked in the zulip chat.

Reply:
Done